Detectors/ITSMFT/ITS/workflow/src/DCSAdaposParserSpec.cxx

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   DCSAdaposParserSpec.cxx
+
+#include "ITSWorkflow/DCSAdaposParserSpec.h"
+
+namespace o2
+{
+namespace its
+{
+
+//////////////////////////////////////////////////////////////////////////////
+// Default constructor
+ITSDCSAdaposParser::ITSDCSAdaposParser()
+{
+  this->mSelfName = o2::utils::Str::concat_string(ChipMappingITS::getName(), "ITSDCSAdaposParser");
+}
+
+//////////////////////////////////////////////////////////////////////////////
+void ITSDCSAdaposParser::init(InitContext& ic)
+{
+  LOGF(info, "ITSDCSAdaposParser init...", mSelfName);
+
+  this->mCcdbUrl = ic.options().get<std::string>("ccdb-url");
+
+  this->mVerboseOutput = ic.options().get<bool>("verbose");
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Main running function
+// Get Data from ADAPOS and prepare CCDB objects
+void ITSDCSAdaposParser::run(ProcessingContext& pc)
+{
+
+  // Retrieve adapos data and process them
+  auto dps = pc.inputs().get<gsl::span<DPCOM>>("input");
+  process(dps);
+
+  if (doStrobeUpload && dps.size() > 0) {
+    // upload to ccdb
+    pushToCCDB(pc);
+  }
+
+  if (mTF % 50 == 0) { // clean memory after some TFs
+    mDPstrobe.clear();
+  }
+
+  mTF++;
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Function to process DPs
+void ITSDCSAdaposParser::process(const gsl::span<const DPCOM> dps)
+{
+
+  // first we check which DPs are missing - if some are, it means that
+  // the delta map was sent
+  if (mVerboseOutput) {
+    LOG(info) << "\n\n\nProcessing new TF\n-----------------";
+  }
+
+  // Process all DPs, one by one
+  for (const auto& it : dps) {
+    processDP(it);
+  }
+
+  /**************************************
+     decide whether to upload or not the object to ccdb. The logic for strobe length is:
+     - if the strobe length it's the first value filled in mDPstrobe, then upload it to CCDB (only at the first call of the wf)
+     - if the strobe length it's the second value filled, compare it with the previous one: if different, upload the most recent
+     - the logic continue... memory is cleaned every 50 TFs (random number of TFs).
+  ***************************************/
```

Comment:
why don't you fill the new value only when the value provided by adapos is different from the one read from the CCDB at the beginning of the workflow?

Not sure what this 50 stands for? The "TFs" seen by the DCS DP workflow have nothing to do with run TFs. 
The ADAPOS supplies full image map once per a few seconds and in between it supplies deltas.  In case ITS is subscribed to some other DPs, there might be plenty of "TFs" with deltas shipped.

Reply:
@shahor02 I could try to read the ccdb object at the beginning of `run()` so that every time it reads a potentially new object. I guess in this case I have to use the `BasicCCDBManager` right? 

The 50 there is just a random number, I accumulate 50 DPs per stave and then I clean the memory.